clientconn.go

```diff
@@ -196,7 +195,7 @@ func WithTimeout(d time.Duration) DialOption {
 }
 
 // WithDialer returns a DialOption that specifies a function to use for dialing network addresses.
-func WithDialer(f func(addr string, timeout time.Duration) (net.Conn, error)) DialOption {
+func WithDialer(f transport.DialerFn) DialOption {
```

Comment:
Please do not expose transport package to grpc users. We intend to make transport as a private package which is only used by grpc package.